Added the STEP and WRL Files

Electro707 opened the pull request:
Re-submission of pull request #628  as I forgot to create a separate branch for the commit.

Pull request to add the 3D model for a 5mm RGB Wide Pin LED.
Screenshot of part in FreeCAD:
![FreeCAD Image](https://user-images.githubusercontent.com/15680882/71229120-6b6c3c80-22b2-11ea-9f56-5061e13a7e78.png)
Screenshot of part in KiCad:
![KiCad Image](https://user-images.githubusercontent.com/15680882/71229225-b9814000-22b2-11ea-9135-098f0842f0e9.png)

------------
Thanks for creating a pull request to contribute to the KiCad libraries! To speed up integration of your PR, please check the following items as you complete them:

- [x] A screenshot of the model(s) aligned with KiCad footprints is *very useful*
- [x] Was the 3D model(s) created in a mechanical modelling program?
- [x] Source files (or scripts) have been contributed 
- [x] Step model submitted in 1:1 scale
- [x] WRL model submitted in 1:2.54 scale
- [x] 3D model alignment checked against KiCad footprint
- [ ] WRL model uses correct material properties
- [x] Geometry check performed against mechanical models

antoniovazquezblanco commented:
Please, put a link in this thread to the source PR of this model. Thank you!

Electro707 commented:
Here is the link for the source's pull request:
[https://github.com/KiCad/kicad-packages3D-source/pull/254](https://github.com/KiCad/kicad-packages3D-source/pull/254)

Misca1234 commented:
Hi

Thanks for your contribution


You need to open the .step file in a text editor and change the following
"FILE_NAME('F:/Kicad/Parts/Self-Made\\Body001.step'"
into
"FILE_NAME('LED_D5.0mm-4_RGB_Wide_Pins.step'"

and
"#7 = PRODUCT('Body001','Body001','',(#8));"
into
"#7 = PRODUCT('LED_D5.0mm-4_RGB_Wide_Pins','LED_D5.0mm-4_RGB_Wide_Pins','',(#8));"

This is generic model, not necessary only for 
https://www.kingbright.com/attachments/file/psearch/000/00/00/L-154A4SUREQBFZGEW(Ver.12A).pdf

so the measurements are ok 
however you should reconsider not to have thje whole body transparent, 
perhaps only the top, now grey is sos booring so if you change the body into non transparent, choose some fancy colour for it, like blue, red or green so the PCB "lights up" a little in 3D mode.

Electro707 commented:
Changed step file properties as requested. Also changed the LED color to blue:
![image](https://user-images.githubusercontent.com/15680882/72944919-60407c80-3d48-11ea-9758-fc0251d8a111.png)

I have recently soldered some RGB LEDs with the footprint I am making the 3d model for, and I noticed that the LEDs were higher than modeled (due to the fact I just pushed the pins thru instead of individually bending each pin). Should I change the 3d model so the LED is elevated some more from the PCB surface (So it models how most RGB LEDs will be soldered with this footprint)?

Electro707 commented:
Changed the LED to this:
![image](https://user-images.githubusercontent.com/15680882/76910438-476ac800-68ae-11ea-87b2-8181e6d555c6.png)
![image](https://user-images.githubusercontent.com/15680882/76910532-87ca4600-68ae-11ea-93df-d20919f41fff.png)
The main reason is to change the way it was modeled in FreeCad (Now it uses an additive pipe to extrude the legs). Also the way i modeled it now makes the LED 2mm off the PCB base, which is more realistic than before I think.

Electro707 commented:
Any updates on this pull request ?

Franck78 commented:
Hello @Electro707

The footprint is quite simple : 4 holes  spaced 1.27 so that the LED fits in without bending anything.

This drawing have zero chance to be accepted for the model described in the PDF !

Also, I have opened your model with Freecad 0.18 . Nothing can be done with the file. Too old FreeCad ?

Fix the Led with straight legs

![kbled](https://user-images.githubusercontent.com/1778884/94315387-b5412c00-ff82-11ea-8344-1ce1280a55e6.png)

Electro707 commented:
@Franck78 The footprint that you're referring to is the _LED_D5.0mm-4_RGB_, whereas this PR is for the _LED_D5.0mm-4_RGB_Wide_Pins_ package, which as the pins spaced at 2.16mm apart (for easier hand-soldering assembly, as I have made PCBs with this footprint beforehand). 
![image](https://user-images.githubusercontent.com/15680882/94330986-33a7c900-ff97-11ea-9399-cc51b27f5888.png)
As for the FreeCAD files, it opens fine on FreeCAD 0.18.4. You might need to install the kicad-stepup add-on if you don't have it?
![image](https://user-images.githubusercontent.com/15680882/94330946-eaf01000-ff96-11ea-8736-b56738d67a9e.png)

Franck78 commented:
ok, that  is a new one for me (bending ping manually).

I have StepUp installed. It is Freecad just a little silly. As the model is saved, when you open any sketch, nothing is displayed but constraint properties and items lists, yes. A bit weird.  For me, the fusioned model is useless and the interesting things are the parts (the visibility is inverted). Now I know ;)

If you made the 'straight & staggered' and this one from the same source, this should be validated.

Strange you did not 'suppress' the extra constraint error.

![ledrgb](https://user-images.githubusercontent.com/1778884/94337887-1f42eb00-ffee-11ea-8942-d8aad106f593.png)
